tooling/bundler/src/bundle/windows/templates/installer.nsi

```diff
@@ -744,15 +751,22 @@ Function RestorePreviousInstallLocation
     StrCpy $INSTDIR $4
 FunctionEnd
 
+Function Skip
+  Abort
+FunctionEnd
+
 Function SkipIfPassive
   ${IfThen} $PassiveMode = 1  ${|} Abort ${|}
 FunctionEnd
 
 Function CreateOrUpdateStartMenuShortcut
   ; We used to use product name as MAINBINARYNAME
   ; migrate old shortcuts to target the new MAINBINARYNAME
-  ${If} ${FileExists} "$DESKTOP\${PRODUCTNAME}.lnk"
-    !insertmacro SetShortcutTarget "$DESKTOP\${PRODUCTNAME}.lnk" "$INSTDIR\${MAINBINARYNAME}.exe"
+  ${If} ${FileExists} "$SMPROGRAMS\$AppStartMenuFolder\${PRODUCTNAME}.lnk"
+    !insertmacro SetShortcutTarget "$SMPROGRAMS\$AppStartMenuFolder\${PRODUCTNAME}.lnk" "$INSTDIR\${MAINBINARYNAME}.exe"
+    Return
+  ${ElseIf} ${FileExists} "$SMPROGRAMS\${PRODUCTNAME}.lnk"
+    !insertmacro SetShortcutTarget "$SMPROGRAMS\${PRODUCTNAME}.lnk" "$INSTDIR\${MAINBINARYNAME}.exe"
```

Comment:
shouldn't we upgrade both if they exist? and not just update one or the other?

Reply:
Migration should only need to happen once, and unless they had both of them before, there're no reason to check and migrate both

Reply:
after this PR, developers may change the start menu folder back and forth if they wish.

Reply:
It shouldn't happen at all, the first time user installs it, it migrates the existing one, after the first time, we won't need the migration code to work anymore

Reply:
the migration code will always run on update, because if you changed the main binary name, we still need to update the shortcut

Reply:
it will always run on update, but it won't matter if it works or not after the first run

Reply:
not sure I understand what you mean

Reply:
if we are updating one shortcut, why only that one, we can also update the other one as well

Reply:
I don't think it can happen, but if you're concerned I'll add it